src/communication/NetworkClient.ts

```diff
@@ -161,28 +145,76 @@ export default class NetworkClient {
     return embedded[binderName] as R[];
   }
 
-  iterate<R>(relativePath: string, type: string, query: Record<string, any> = {}): HelpfulIterator<R> {
+  iterate<R>(pathname: string, binderName: string, query: Maybe<SearchParameters>, valuesPerMinute = 5000): HelpfulIterator<R> {
     return new DemandingIterator(demand => {
-      // Pick a limit (page size) based on the guessed demand. (The magic numbers below: 128 is the limit used if no
-      // demand was guessed; 250 is the maximal limit imposed by the Mollie API; 64 is the minimal limit, to ensure
-      // inaccurate guesses do not result in wastefully short pages.)
-      let limit: number;
+      // Pick the page sizes (limits) based on the guessed demand.
+      let popLimit: () => number;
+      // If no demand was guessed, pages of 128 values each will be requested.
       if (demand == Number.POSITIVE_INFINITY) {
-        limit = 128;
+        popLimit = () => 128;
       } /* if (demand != Number.POSITIVE_INFINITY) */ else {
-        limit = Math.max(Math.ceil(demand / Math.ceil(demand / 250)), 64);
+        let pageSizes: number[];
+        // If a demand of 64 or less was guessed, request an initial page of 64 values. This ensures inaccurate guesses
+        // do not result in wastefully short pages.
+        if (demand <= 64) {
+          pageSizes = [64];
+        } /* if (demand > 64 && demand != Number.POSITIVE_INFINITY) */ else {
+          // If a demand of over 64 was guessed, request pages of 250 values ‒ a limit imposed by the Mollie API ‒
+          // until the remaining demand can be met with a single page. Finally, request that single page.
+          pageSizes = new Array(Math.ceil(demand / 250));
+          pageSizes[0] = demand - (pageSizes.length - 1) * 250;
+          pageSizes.fill(250, 1);
+        }
+        // If more values are requested after the guessed demand (the guess was inaccurate), request pages of 128
+        // values each.
+        popLimit = () => pageSizes.pop() ?? 128;
       }
-      return new HelpfulIterator<R>(iterate(this.axiosInstance, type, `${relativePath}${stringifyQuery({ ...query, limit })}`));
+      // Set up the throttler. The iterator will wait before making a request to the Mollie API if more values have
+      // been consumed than "allowed" by the valuesPerMinute argument. Note that this iterator will not interrupt the
+      // yielding of values which have already been received from the Mollie API. If a page of 250 values is received
+      // and valuesPerMinute is set to 100, all 250 values received values will be yielded before the (two-minute)
+      // break.
+      const throttler = new Throttler(valuesPerMinute);
```

Comment:
I may have overlooked this, but are we testing the throttling logic specifically?

Reply:
We're not testing the timing specifically, no. That would be nice…

Reply:
I added a throttling test. As a nice side-effect, it replaces the previous "multipage" test with a much simpler implementation.